bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/GoldenRecordUpdateServices.kt

```diff
@@ -307,13 +306,34 @@ class GoldenRecordUpdateChunkService(
         businessPartner.siteConfidence?.let { update(it,  site.confidenceCriteria) }
     }
 
-    private fun update(businessPartner: BusinessPartnerDb, address: LogisticAddressVerboseDto){
+    private fun update(businessPartner: BusinessPartnerDb, address: LogisticAddressVerboseDto) : OutputUpsertData{
         updateIdentifiers(businessPartner.identifiers, address.identifiers.map(::toEntity), BusinessPartnerType.ADDRESS)
         updateStates(businessPartner.states, address.states, BusinessPartnerType.ADDRESS)
         businessPartner.addressName = address.name
         businessPartner.postalAddress.physicalPostalAddress = address.physicalPostalAddress.toEntity()
         businessPartner.postalAddress.alternativePostalAddress = address.alternativePostalAddress?.toEntity()
         businessPartner.addressConfidence?.let { update(it,  address.confidenceCriteria) }
+
+        //Below code will be used when,
+        //When addressType has been changed from LegalAddress to LegalAndSiteMainAddress
+        //When addressType has been changed from AdditionalAddress to SiteMainAddress
+        if (address.bpnSite != null && businessPartner.bpnS == null) {
+            businessPartner.bpnS = address.bpnSite
+            val searchRequest = SiteSearchRequest(siteBpns = listOf(address.bpnSite!!))
+            val sites = if(searchRequest.siteBpns.isNotEmpty())
+                poolClient.sites.getSites(searchRequest, PaginationRequest(size = searchRequest.siteBpns.size)).content.map { it.site }
+            else
+                emptyList()
+            //as addressType has been updated, so there is no siteConfidence criteria.
+            //fill the fake confidence to prevent the error which will update in further steps.
+            businessPartner.siteConfidence = ConfidenceCriteriaDb(false,false,0, LocalDateTime.now(), LocalDateTime.now(),0)
```

Comment:
In future, we need to change LocalDateTime to Instant on entity level but not for now.